### PR TITLE
Convert kola test jobs to vanilla pipeline jobs

### PR DESCRIPTION
**Pull request by jlebon**
This is part of the push to move away from buildconfig-based jobs
towards the same model as what upstream CI does. For more details, see:

https://github.com/coreos/fedora-coreos-pipeline/pull/228

As a first victim, convert the kola tests since they're mostly
straight-forward. A huge part of this is re-using coreos-ci-lib the same
way upstream CI does. And this helps save us a lot of boilerplate (see
diffstat for this patch).

Converting to vanilla jobs also means we can now use the `build` step as
intended, instead of hacking around it with `oc start-build`.

### CODE REVIEW

**Comment by jlebon** [comment]
Requires: https://github.com/coreos/coreos-ci-lib/pull/36
Requires: https://github.com/coreos/coreos-ci-lib/pull/37

**Comment by jlebon** [comment]
(I'll admit I didn't test this. Since it's not the main pipeline itself, was thinking of just doing fixups as needed on top.)

File: jobs/kola-gcp.Jenkinsfile
Comment (miabbott) [inline jobs/kola-gcp.Jenkinsfile:48]: maybe in the future this can be a function in coreos-ci-lib that can be reused

**Comment by miabbott** [comment]
Looks pretty straight-forward to me.  :+1:

File: jobs/kola-gcp.Jenkinsfile
Comment (jlebon) [reply jobs/kola-gcp.Jenkinsfile:48]: Yeah, makes sense. Maybe an `fcosBuildPrep` or something? I think the f-c-c CI could use this as well.

File: Jenkinsfile
Comment (dustymabe) [inline Jenkinsfile:553]:
```suggestion
                // We consider the GCP kola tests to be a followup job, so we use `wait: false` here.
```

File: jobs/kola-aws.Jenkinsfile
Comment (dustymabe) [inline jobs/kola-aws.Jenkinsfile:22]:
```suggestion
             description: 'Optional: Override the Fedora CoreOS S3 Stream Directory',
```

File: jobs/kola-aws.Jenkinsfile
Comment (dustymabe) [inline jobs/kola-aws.Jenkinsfile:45]:
```suggestion
        export AWS_CONFIG_FILE=\${AWS_FCOS_BUILDS_BOT_CONFIG}
```

Is what we had in the past. Shall we keep it the same in case something changes we won't need to update this code, just the variable.

File: jobs/kola-aws.Jenkinsfile
Comment (dustymabe) [reply jobs/kola-aws.Jenkinsfile:22]: Same suggestion in GCP file

File: jobs/kola-aws.Jenkinsfile
Comment (dustymabe) [reply jobs/kola-aws.Jenkinsfile:45]: Same suggestion in GCP file

File: jobs/kola-gcp.Jenkinsfile
Comment (dustymabe) [inline jobs/kola-gcp.Jenkinsfile:65]: any way to break this up into multiple lines? tends to make it easier to read when we have the `--` options on multiple lines like they were before.

File: jobs/kola-aws.Jenkinsfile
Comment (dustymabe) [inline jobs/kola-aws.Jenkinsfile:61]: how are the credentials getting passed in here?

previously we had `export AWS_CONFIG_FILE=\${AWS_FCOS_KOLA_BOT_CONFIG}` right before we called `cosa kola run`.

For GCP we pass in `--gce-json-key=` so that's taken care of.

**Comment by jlebon** [comment]
Updated this now, but I'd like to sanity-check those `kola run` invocations locally.

File: jobs/kola-aws.Jenkinsfile
Comment (jlebon) [reply jobs/kola-aws.Jenkinsfile]: Yup, good catch!

File: jobs/kola-aws.Jenkinsfile
Comment (dustymabe) [inline jobs/kola-aws.Jenkinsfile:42]: hmm. That's not exactly what I had in mind. I think I'm bumping into a limitation in my knowledge on how things work.

I was thinking we'd be somehow re-using 

https://github.com/coreos/fedora-coreos-pipeline/blob/35f1f18598cbe39601d71f5a181248288e6357bc/manifests/pod.yaml#L23-L24

rather than having to define it here again.

Comment (jlebon) [reply jobs/kola-aws.Jenkinsfile:42]: We no longer use that pod definition in those jobs. Instead, we're telling `cosaPod` that we need access to the specific secrets (see https://github.com/coreos/coreos-ci-lib/pull/37). That way, pods only get the secrets they need mounted in.

I guess an improvement we could make here is to expose the path to the secret in environment variables as part of that wrapper.

Comment (jlebon) [reply jobs/kola-aws.Jenkinsfile:42]: E.g. in this file, notice:

```groovy
cosaPod(image: params.COREOS_ASSEMBLER_IMAGE, memory: "256Mi",
        secrets: ["aws-fcos-builds-bot-config", "aws-fcos-kola-bot-config"]) {
```

Comment (dustymabe) [reply jobs/kola-aws.Jenkinsfile:42]: OK, sorry for the noise on this. I probably wouldn't have made the [other comment](https://github.com/coreos/fedora-coreos-pipeline/pull/270#discussion_r494999308) if I had realized how it worked.

**Review by dustymabe — APPROVED** [review]
LGTM

File: jobs/kola-aws.Jenkinsfile
Comment (jlebon) [reply jobs/kola-aws.Jenkinsfile:42]: No worries, I think a good idea came out of it regardless: https://github.com/coreos/coreos-ci-lib/pull/39.

**Review by cgwalters — APPROVED** [review]
I am not an expert in this pipeline, but the high level direction looks really good to me, and superficially code looks good too.

**Comment by darkmuggle** [comment]
LGTM....any thing hold up the merge?

**Comment by dustymabe** [comment]
Maybe let's hold just a day or two til we get the `next` release out the door and then merge away!

**Comment by jlebon** [comment]
OK, manually tested this locally now and fixed a bunch of issues. Also realized there's a gap in `fcosKola` we need to address first: https://github.com/coreos/coreos-ci-lib/pull/41. Once that PR is merged, I think we can merge this.

**Review by dustymabe — APPROVED** [review]
LGTM

**Pull request merged**
